### PR TITLE
add dagger CoroutinesModule

### LINKED ISSUE #13: CoroutinesDispatchers via Dagger
main, IO, default singleton dispatchers.

### PR DESCRIPTION
**Pull request by zhaonian**
Put all 3 dispatchers in the `CoroutinesModule` and added this `CoroutinesModule` into `AppComponent` to make it app-wide.

TESTED=dummy function in `AuthViewModel` to log the messages.

Fix https://github.com/zhaonian/GHub/issues/13

### CODE REVIEW

File: app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt [hunk 1]
```diff
@@ -1,6 +1,25 @@
 package io.zluan.ghub.ui.auth
 
 import androidx.lifecycle.ViewModel
+import io.zluan.ghub.di.coroutines.IODispatcher
+import kotlinx.coroutines.*
```
Comment (jaredsburrows) [inline app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt]: Import everything? :P

File: app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+import timber.log.Timber
 import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
 
-class AuthViewModel @Inject constructor() : ViewModel()
+class AuthViewModel @Inject constructor(
```
Comment (jaredsburrows) [inline app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt:15]: Add kdoc?

File: app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt (hunk 1 above)
Comment (zhaonian) [reply app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt]: 😈

File: app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt (hunk 2 above)
Comment (zhaonian) [reply app/src/main/java/io/zluan/ghub/ui/auth/AuthViewModel.kt:15]: Done.

**Pull request merged**